### PR TITLE
A proposal for line length limits in the PHP guidelines

### PR DESCRIPTION
**Pull request by garthwebb**
I believe many people follow line length limits at Wikia so it would be nice to formalize these a bit.  This pull request adds guidelines addressing two issues around long lines.  First how long a line should be in general and second what to do about long conditional statements specifically.

Note that this pull request is based on the "Reformat" branch.  Therefore this pull request is dependent on that branches pull request (https://github.com/Wikia/guidelines/pull/59).  This was done to separate these two changes and hopefully doesn't introduce more confusion.  If both pull requests are accepted, this pull request should be pulled first.

@Wikia/engineers

### CODE REVIEW

File: PHP/CodingConventions.md [hunk 1]
```diff
@@ -24,11 +26,66 @@ found in the [MediaWiki Coding Conventions](http://www.mediawiki.org/wiki/Manual
 
 Functions should have a limited length.  This limit is loosely defined as one “page”, where a page is the number
 of lines that fit into your editor window or less.  This definition is intentionally vague to allow flexibility but the
-spirit of this guideline is that a developer should not have to scroll to see a full function definition.
+spirit of this guideline is that a developer should not have to scroll vertically to see a full function definition.
```
Comment (armonr) [inline PHP/CodingConventions.md]: horizontally?

Comment (armonr) [inline PHP/CodingConventions.md]: Oh I see, function's definition, not declaration :)

File: PHP/CodingConventions.md [hunk 2]
```diff
@@ -24,11 +26,66 @@ found in the [MediaWiki Coding Conventions](http://www.mediawiki.org/wiki/Manual
...
+ 
+* each check should be split into it's own line
+* the single conditional should be split into separate conditionals of two or less checks
+* all checks replaced by a function call
+
+Example:
+
+```php
+// Bad
+if ( F::App()->wg->User->can('edit') && F::App()->wg->Skin->getSkinName() == 'oasis' && empty( F::app->wg->NoExternals ) {
```
Comment (jsutterfield) [inline PHP/CodingConventions.md]: We should make sure these examples follow our white space guidelines as well. ```F::App()->wg->User->can('edit')``` should be ```F::App()->wg->User->can( 'edit' )```

File: PHP/CodingConventions.md [hunk 3]
```diff
@@ -24,11 +26,66 @@ found in the [MediaWiki Coding Conventions](http://www.mediawiki.org/wiki/Manual
...
+// Good - each check split into it's own line
+if ( F::App()->wg->User->can('edit') &&
+     F::App()->wg->Skin->getSkinName() == 'oasis' &&
+     empty( F::app->wg->NoExternals )
+   ) {
+    // do something
+}
+
+// Good - single conditional split into separate conditionals of two or less checks
+if ( !F::App()->wg->User->can('edit') ) {
```
Comment (jsutterfield) [inline PHP/CodingConventions.md]: also here and on line 67.

**Comment by lizlux** [comment]
:+1:

File: PHP/CodingConventions.md [hunk 4]
```diff
@@ -24,11 +26,66 @@ found in the [MediaWiki Coding Conventions](http://www.mediawiki.org/wiki/Manual
...
+```php
+// Bad
+if ( F::App()->wg->User->can('edit') && F::App()->wg->Skin->getSkinName() == 'oasis' && empty( F::app->wg->NoExternals ) {
+   // do something
+}
+
+// Good - each check split into it's own line
+if ( F::App()->wg->User->can( 'edit' ) &&
+     F::App()->wg->Skin->getSkinName() == 'oasis' &&
+     empty( F::app->wg->NoExternals )
```
Comment (Grunny) [inline PHP/CodingConventions.md]: `F::app->wg` -> `F::app()->wg`.

File: PHP/CodingConventions.md [hunk 5]
```diff
@@ -24,11 +26,66 @@ found in the [MediaWiki Coding Conventions](http://www.mediawiki.org/wiki/Manual
...
+// Good - checks replaced by a function call
+if ( oasisUserCanEdit() ) {
+    // do something
+}
+
+function oasisUserCanEdit() {
+    return (
+        F::App()->wg->User->can( 'edit' ) &&
+        F::App()->wg->Skin->getSkinName() == 'oasis' &&
+        empty( F::app->wg->NoExternals )
```
Comment (Grunny) [inline PHP/CodingConventions.md]: `F::app->wg` -> `F::app()->wg`.

File: PHP/CodingConventions.md (hunk 2 above)
Comment (Grunny) [inline PHP/CodingConventions.md]: `F::app->wg` -> `F::app()->wg`. And if we're nitpicking on the examples, `F::App()->wg->User->can( 'edit' )` -> `F::app()->wg->User->isAllowed( 'edit' )`. ;)

File: PHP/CodingConventions.md [hunk 6]
```diff
@@ -24,11 +26,66 @@ found in the [MediaWiki Coding Conventions](http://www.mediawiki.org/wiki/Manual
...
+of this guideline is that a developer should not have to scroll horizontally to see the end of a line.
+
+If a single line becomes too long, newlines should be added at appropriate breakpoints.
+
+### Conditional Logic
+
+Conditional tests should not do more than two checks on a single line.  If a conditional requires more than two checks,
+one of the following remedies should be used:
+ 
+* each check should be split into it's own line
```
Comment (Grunny) [inline PHP/CodingConventions.md]: "it's" -> "its", and "onto" is probably better than "into".

File: PHP/CodingConventions.md (hunk 2 above)
Comment (garthwebb) [inline PHP/CodingConventions.md]: Heh, I wish I could remember where I lifted these lines in our codebase so we could fix them ;)

**Comment by Grunny** [comment]
LGTM.

File: PHP/CodingConventions.md [hunk 7]
```diff
@@ -24,11 +26,66 @@ found in the [MediaWiki Coding Conventions](http://www.mediawiki.org/wiki/Manual
...
+ 
+* each check should be split onto its own line
+* the single conditional should be split into separate conditionals of two or less checks
+* all checks replaced by a function call
+
+Example:
+
+```php
+// Bad
+if ( F::App()->wg->User->can('edit') && F::App()->wg->Skin->getSkinName() == 'oasis' && empty( F::app()->wg->NoExternals ) {
```
Comment (lizlux) [inline PHP/CodingConventions.md]: did we decide to fix this `('edit')` whitespace? (I know it's the "bad" example, but still...)

**Pull request merged**